src/screens/home/Home.tsx

```diff
@@ -1,67 +1,99 @@
 import React, { useState } from 'react';
 import { useEffect } from 'react';
-import { View, StyleSheet, Dimensions } from 'react-native';
+import { View, StyleSheet, useWindowDimensions } from 'react-native';
 import MasonryList from 'react-native-masonry-list';
 
 const state = {
   onLoading: true,
   offset: 0,
 };
 
-const screenWidth = Dimensions.get('window').width;
+// [1] Since you're using Functional Components, refrain from using the static
+// `Dimensions.get()` function. Use the `useWindowDimensions` hook instead. 
+// (Refer to the comment section also marked with [1] below.)
 
 const numColumns = 2;
 
-const tileSize = screenWidth / numColumns;
+// [4] What purpose does this global array serves? If it's a temporary variable,
+// define it in the scope. If it is a global variable, one should have a very
+// strong reason to use it. Generally, avoid global singletons as much as you
+// can. Especially those typed with `any`.
+// (Refer to the comment section also marked with [4] below.)
+// let tmp: any[] = [];
 
-const styles = StyleSheet.create({
-  view: {
-    flex: 1,
-    alignItems: 'center',
-    padding: 10,
-  },
-  textInput: {
-    margin: 10,
-    width: '100%',
-    height: 50,
-    color: 'white',
-  },
-  image: {
-    width: tileSize,
-    height: 150,
-    marginBottom: 5,
-  },
-  flatList: {
-    width: screenWidth,
-  },
-});
+// [2] We should define a type for the GIFs instead of using `any`. In our code,
+// we try to avoid using `any` as much as we can, because if we use `any`, then
+// it is no better than just using JavaScript (sometimes could be worse because
+// VS Code can still infer the type using IntelliSense). If you are unfamiliar
+// with the type of the object, you may use Postman to send a test query.
+// (Refer to the comment section also marked with [2] below.)
+interface GiphyTrendingResponse {
+  data: Array<{
+    images: {
+      original: {
+        url: string
+      }
+    }
+  }>
```

Comment:
Notice that here I'm declaring the type for the array using the `Array<type>` syntax instead of `type[]` syntax. Generally, they are equivalent in meaning and is unambiguous. However, in this case, the element type of the array is a very complex object (as you can see, nesting multiple levels). What happens if I define it using the `type[]` syntax?
```typescript
interface GiphyTrendingResponse {
  data: {
    images: {
      original: {
        url: string
      }
    }
  }[]
}
```
At a brief glance, people might thing that data is an object type instead of an array. They have to read all the way to the end of the object to find out that it is in fact an array type of the preceding element type. That's why, I opted for the unambiguous `Array<>` syntax